### PR TITLE
console: allow manual edit of column types and handle array data types (close #2544, #3335, #2583)

### LINKED ISSUE #2544: handle array data types in console
Currently the console doesn't handle array type columns properly. 

Need to add support for:
- create array type columns
- allow modifying array type columns ie: array type casts
- display type of array type columns as `type[]` (eg: `int[]`) everywhere

### PR DESCRIPTION
**Pull request by beerose**
Solves #2544, #3335, https://github.com/hasura/graphql-engine/issues/2583

### Description

Kapture 2020-05-04 at 20 59 58

### Changelog

- [x] `CHANGELOG.md` is updated with user-facing content relevant to this PR.

### Affected components

- [x] Console

### CODE REVIEW

**Comment by rikinsk** [comment]
Will also solve #3335

**Comment by beerose** [comment]
Will also solve https://github.com/hasura/graphql-engine/issues/2583

**Comment by AkshayCloudAnalogy** [comment]
Any update on how we can use array and what will be the type of field for it in hasura?
https://hasura.io/docs/1.0/graphql/manual/mutations/insert.html#insert-an-object-with-an-array-field
This is not working and sending back as string and also there is no text[] or _text in field types in hasura.

**Comment by dohomi** [comment]
I am using currently `TEXT` with `{stringA,stringB}` fields inside of project. Will this be backwards compatible change?

**Comment by beerose** [comment]
@dohomi Yes, it will be backward compatible change.

File: console/src/components/Common/SearchableSelect/SearchableSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+type Option = { label: string; value: string };
+
+const CustomOption: React.FC<OptionProps<OptionTypeBase>> = props => {
+  return (
+    <div
+      title={props.data.description || ''}
+      data-test={`data_test_column_type_value_${props.data.value}`}
+    >
+      <components.Option {...props} />
```
Comment (soorajshankar) [inline console/src/components/Common/SearchableSelect/SearchableSelect.tsx:20]: can we have this parsed  (for better readability)
```jsx
const {Option}=components;
...
 <Option {...props} />
```

File: console/src/components/Common/SearchableSelect/SearchableSelect.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    </div>
+  );
+};
+
+type Props = {
```
Comment (soorajshankar) [inline console/src/components/Common/SearchableSelect/SearchableSelect.tsx:25]: can we rename this to `SearchableSelectProps` and export this?

File: console/src/components/Services/Data/Common/Components/ColumnTypeSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+type Option = {
+  value: string;
+  label: string;
+  description?: string;
+  key: number;
+  colIdentifier: number;
+};
+
+type Props = {
```
Comment (soorajshankar) [inline console/src/components/Services/Data/Common/Components/ColumnTypeSelector.tsx:13]: same^

File: console/src/components/Services/Data/TableModify/ColumnEditor.js [hunk 4]
```diff
@@ -30,7 +30,10 @@ const ColumnEditor = ({
       selectedProperties[colName].type
     );
   };
-  const columnTypePG = getColumnType();
+  let columnTypePG = getColumnType();
+  if (columnProperties.display_type_name === 'ARRAY') {
```
Comment (soorajshankar) [inline console/src/components/Services/Data/TableModify/ColumnEditor.js:34]: can we reuse the constant `ARRAY` here?

File: console/src/components/Services/Data/TableModify/ColumnEditorList.js [hunk 5]
```diff
@@ -59,14 +59,28 @@ const ColumnEditorList = ({
    * */
   return columns.map((col, i) => {
     const colName = col.column_name;
+    const isArrayDataType = col.data_type === 'ARRAY';
```
Comment (soorajshankar) [inline console/src/components/Services/Data/TableModify/ColumnEditorList.js:62]: ^ same

**Pull request merged**